src/Service/VendorService/OpenLibrary/OpenLibraryVendor.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Service\VendorService\OpenLibrary;
+
+use App\Exception\UnsupportedIdentifierTypeException;
+use App\Service\VendorService\VendorServiceSingleIdentifierInterface;
+use App\Service\VendorService\VendorServiceTrait;
+use App\Utils\CoverVendor\UnverifiedVendorImageItem;
+use App\Utils\Types\IdentifierType;
+
+class OpenLibraryVendor implements VendorServiceSingleIdentifierInterface
+{
+    use VendorServiceTrait;
+
+    private const VENDOR_ID = 20;
+
+    /**
+     * Example: https://covers.openlibrary.org/b/isbn/9780385472579-L.jpg.
+     *
+     * @see https://openlibrary.org/dev/docs/api/covers
+     */
+    private const COVER_URL_FORMAT = 'https://covers.openlibrary.org/b/isbn/%s-L.jpg?default=false';
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getUnverifiedVendorImageItem(string $identifier, string $type): UnverifiedVendorImageItem
+    {
+        if (!$this->supportsIdentifierType($type)) {
+            throw new UnsupportedIdentifierTypeException('Unsupported single identifier type: '.$type);
+        }
+
+        $item = new UnverifiedVendorImageItem();
+        $item->setIdentifier($identifier);
+        $item->setIdentifierType($type);
+        $item->setVendor($this->vendorCoreService->getVendor(self::VENDOR_ID));
+        $item->setOriginalFile($this->getVendorsImageUrl($identifier));
+
+        return $item;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function supportsIdentifierType(string $type): bool
+    {
+        return IdentifierType::ISBN === $type;
+    }
+
+    /**
+     * Get Vendors image URL from ISBN.
+     */
+    private function getVendorsImageUrl(string $isbn): string
+    {
+        return sprintf(self::COVER_URL_FORMAT, $isbn);
```

Comment:
Should we look into using https://github.com/thecodingmachine/safe ?

`use function safe/sprintf as sprintf`

Reply:
Changelog:
PHP 8: This function no longer returns false on failure. 
https://www.php.net/manual/en/function.sprintf.php and https://github.com/thecodingmachine/safe/blob/d48e85e101330b30b0c2555df063f0c719b98d7f/deprecated/strings.php#L296

So no reason to introduce `thecodingmachine/safe` here. But we can consider it for a later time.